### DISCUSSION TITLE
基于InternLM的agent应用——租房小助手（选题参考）（已满）

### DISCUSSION THREAD

**Discussion by sanbuphy** [InternLM/Tutorial#384] [project]
## 背景

租房是一个常见的生活需求，但在庞大的房屋市场中找到理想的租房信息并不容易。为了解决这一问题，我们开发了一个基于InternLM的Agent应用——租房小助手。

租房小助手利用InternLM的强大语言模型和联网功能，为用户提供个性化、高效的租房服务。它可以帮助用户快速搜索、筛选和比较各种租房选项，从而节省时间和精力。

主要功能和特点：

- 根据用户输入进行单次或者多轮询问，构建基于房子的各类特征 json 结构化信息（比如说这个地段有什么，然后电力多少，价格如何），可以直接存储用于下次比对

- 根据常见租房需要，构建一系列需要注意的要点（坑点），询问用户，然后存储信息或给出建议（比如提醒用户需要注意XXXX，然后基于用户的回答去构建回复）

- 根据常见租房问题，构建一系列问题（agent需要对这个进行选择或打分，或者给出建议）

通过租房小助手，所有想去租房的应届毕业生可以更加便捷地找到符合自己需求的租房选项，节省时间和精力，同时提高租房决策的准确性。让我们一起提高全体普通人的生活幸福指数，降低信息差、降低获取幸福的成本。

## 技术栈

可参考  lagent、**metagpt**  ；metagpt 可以直接更换接口后得到结果，lagent之后会有更新

**Comment by zhjunqin** [comment]
想报名这个，zhjunqin

请问一下，这种有相关租房查询 API 接口可以调用么

**Reply by sanbuphy** [reply]
联系助教  然后让他帮忙拉群 喊话即可。

这个其实咱们可以找找

**Comment by ljqiff** [comment]
我最近也在研究agent  加我一个

**Reply by sanbuphy** [reply]
你好 目前人数已满

**Comment by VinciLee1** [comment]
非常有需求！我最近因为在找房源，又不想通过中介，所以想构建这样的agent。目前我的思路是：通过crawler获取房源信息，这些已经有很多项目了，比如[豆瓣crawler](https://github.com/topics/douban-crawler)，或者[小红书crawler](https://github.com/NanmiCoder/MediaCrawler/tree/main)。接着，一些地理位置和价格信息等，需要通过gpt进行结构化，然后存储到数据库。最后，利用高德地图的api，通过地理位置实现附近区域，路线规划的查询。
故此，能满足我对于想要找地铁沿线，不超过1km，价格在某一区间的需求。我正在构建这个系统。希望寻求进一步合作。

**Comment by ljqiff** [comment]
已收到你的邮件。
